library/std/src/rt.rs

```diff
@@ -81,8 +81,12 @@ unsafe fn init(argc: isize, argv: *const *const u8) {
         // Next, set up the current Thread with the guard information we just
         // created. Note that this isn't necessary in general for new threads,
         // but we just do this to name the main thread and to give it correct
-        // info about the stack bounds.
-        let thread = Thread::new(Some(rtunwrap!(Ok, CString::new("main"))));
+        // info about the stack bounds. Normally, `thread::spawn` will call
+        // `Thread::set_name` but since this thread already exists, we have
+        // to call it ourselves.
+        let thread_name = rtunwrap!(Ok, CString::new("main"));
```

Comment:
`prctl(PR_SET_NAME, "main")` on the main thread will change the name of the whole program to main, overriding the actual process name. For example for

```c
#include <sys/prctl.h>

int main(void) {
    prctl(PR_SET_NAME, "main");
    for(;;);
}
```

`ps` shows

```
   3086 pts/2    00:00:19 main
```

and top shows

```
   3086 bjorn     20   0    2172    504    440 R  99,7   0,0   0:33.55 main
```

Same with ksysguard (kde's system monitor) It also shows "main" as name. `htop` is the only program I found that shows the actual executable name or rather the full commandline.

Reply:
Ah, this was [encountered by Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1649010). Their solution was to special case Linux.

Reply:
Well, that's very unexpected. Thanks for pointing that out @bjorn3! 

I think it's still worth doing this for Windows since when debuggers attach to the process, they create some threads for themselves in the process and so any Rust program, even `helloworld`, being debugged usually has several threads, only one of which is relevant to the user. 

Does anyone have an objection to this call being cfg'd to Windows only? I suppose, based on the FF thread @ChrisDenton mentioned, this can also be done on macOS but I don't have access to a Mac at the moment to test. I guess we could also just cfg Linux out specifically but it seems like there could be other platforms with similar behavior so it seems better to only do this for platforms we've tested on.

Reply:
Sounds good. I'd be ok with making this a Windows only PR but I would prefer if we could do this for at least macOS at the same time. So long as there is someone to test it. cc @thomcc maybe?

Reply:
Yeah, sorry, To be clear, I meant would you be willing to test a PR that sets the main thread name. And that you aren't aware of any issues with setting the main thread name on macOS.

Reply:
`pthread_setname_np` should be fine. I don't expect any issues, it shouldn't change the process name either (I think).

Reply:
Code is up and CI is green again! @thomcc thanks for being willing to test on macOS!